### PR TITLE
docs(kic): update observability references for fallback config feature

### PR DESCRIPTION
**Pull request by czeslavo**
### Description

Updates KIC's Kubernetes Events and Prometheus metrics references to account for the Fallback Configuration feature added ones.
 
Part of https://github.com/Kong/kubernetes-ingress-controller/issues/5934.

### Testing instructions

Preview link: https://deploy-preview-7467--kongdocs.netlify.app/kubernetes-ingress-controller/unreleased/production/observability/prometheus/
https://deploy-preview-7467--kongdocs.netlify.app/kubernetes-ingress-controller/unreleased/production/observability/events/

### Checklist 

- [x] Review label added 
- [x] [Conditional version tags](https://docs.konghq.com/contributing/conditional-rendering/#conditionally-render-content-by-version) added, if applicable.

### CODE REVIEW

**Comment by czeslavo** [comment]
Vale detects metrics names to be spelling errors, should I add them individually to `dictionary.txt` or is there a better way to fix that?

File: app/_src/kubernetes-ingress-controller/production/observability/prometheus.md [hunk 1]
```diff
@@ -74,6 +74,78 @@ This metric provides these labels:
...
+
+### ingress_controller_fallback_configuration_push_duration_milliseconds
+
+`ingress_controller_fallback_configuration_push_duration_milliseconds` (type: `histogram`) provides the amount of time, in milliseconds, that it takes to push the fallback configuration to Kong.
+
+This metric provides these labels:
+
+* `dataplane` describes the data plane that was the target of the configuration push.
+* `protocol` describes the configuration protocol in use, which can be `db-less` or `deck`.
+* `success` logs the status of configuration updates. If `success` is `false`, an unrecoverable error occurred. If `success` is `true`, the push succeeded with no errors.
```
Comment (programmer04) [inline app/_src/kubernetes-ingress-controller/production/observability/prometheus.md:121]:
```suggestion
* `success` logs the status of configuration updates. If `success` is `false`, an unrecoverable error occurs. If `success` is `true`, the push succeeded without errors.
```

File: app/_src/kubernetes-ingress-controller/production/observability/prometheus.md [hunk 2]
```diff
@@ -74,6 +74,78 @@ This metric provides these labels:
...
+
+### ingress_controller_fallback_configuration_push_broken_resource_count
+
+`ingress_controller_fallback_configuration_push_broken_resource_count` (type: `gauge`) provides the number of resources not accepted by Kong when attempting to push fallback configuration.
```
Comment (programmer04) [inline app/_src/kubernetes-ingress-controller/production/observability/prometheus.md:125]:
```suggestion
`ingress_controller_fallback_configuration_push_broken_resource_count` (type: `gauge`) provides the number of resources not accepted by Kong when attempting to push the fallback configuration.
```

File: app/_src/kubernetes-ingress-controller/production/observability/prometheus.md [hunk 3]
```diff
@@ -74,6 +74,78 @@ This metric provides these labels:
...
+
+* `dataplane` describes the data plane that was the target of the configuration push.
+
+### ingress_controller_fallback_cache_generation_duration_milliseconds
+
+`ingress_controller_fallback_cache_generation_duration_milliseconds` (type: `histogram`) provides the amount of time, in milliseconds, that it takes to generate a fallback cache.
+
+This metric provides the `success` label:
+
+* `success` logs the status of cache generation. If `success` is `false`, an unrecoverable error occurred. If `success` is `true`, the cache generation succeeded with no errors.
```
Comment (programmer04) [inline app/_src/kubernetes-ingress-controller/production/observability/prometheus.md:137]:
```suggestion
* `success` logs the status of cache generation. If `success` is `false`, an unrecoverable error occurs. If `success` is `true`, the cache generation succeeded without errors.
```

**Comment by Guaris** [comment]
@czeslavo Yes, you can add these entries in alphabetical order to `dictionary.txt`.

**Comment by czeslavo** [comment]
> @czeslavo Yes, you can add these entries in alphabetical order to `dictionary.txt`.

Done ✅

**Review by programmer04 — APPROVED** [review]

**Review by mheap — APPROVED** [review]

**Pull request merged**